File: 55.js
function shortestTime(n, edges, src) {
  const graph = Array.from({ length: n }, () => []);
  for (const [u, v] of edges) {
    graph[u].push(v);
    graph[v].push(u);
  }
  const dist = Array(n).fill(-1);
  dist[src] = 0;
  const q = [src];

  while (q.length) {
    const u = q.shift();
    for (const v of graph[u]) {
      if (dist[v] === -1) {
        dist[v] = dist[u] + 1;
        q.push(v);
      }
    }
  }
  return dist;
}

File: 55.py
from collections import deque
def shortestTime(n, edges, src):
    graph = [[] for _ in range(n)]
    for u,v in edges:
        graph[u].append(v)
        graph[v].append(u)
    dist=[-1]*n
    dist[src]=0
    q=deque([src])
    while q:
        u=q.popleft()
        for v in graph[u]:
            if dist[v]==-1:
                dist[v]=dist[u]+1
                q.append(v)
    return dist
